src/app/tide/theoretical_peak_set.h

```diff
@@ -1,774 +1,131 @@
-// though some subclasses will never write any negative peaks. When present,
-// these negative peaks represent corrections made to the peaks in the positive
-// set. This comes up in cases where the workspace in question actually
-// represents a difference between two TheoreticalPeakSets. See
-// TheoreticalPeakSetDiff for example.
+// Benjamin Diament and Attila Kertesz-Farkas
+
+// Originally implemented by Benjamin Diament. It was reimplemented by Attila 
+// Kertesz-Farkas in December 2021.
+// Benjamin's implementetion used a structure to keep track of the types of the
+// theoretical and the experimental peaks. There were 10 types of peaks such as:
+// PeakMain = 0,
+// LossPeak = 1,
+// FlankingPeak = 2,
+// PrimaryPeak = 3,
+// PeakCombinedB1 = 4, // Represents charge 1 B ion, its flanks and neutral losses.
+// PeakCombinedY1 = 5, // Represents charge 1 Y ion, its flanks and neutral losses.
+// // Each of the charge 2 peak types below includes an 'a' version and a 'b'
+// // version. In the 'a' version the loss of ammonia is 9 Da. per charge.
+// // In the 'b' version the loss of ammonia is 8 Da. per charge.
+// PeakCombinedB2 = 6, // Represents charge 2 B ion, its flanks and neutral losses.
+// PeakCombinedY2 = 7, // Represents charge 2 Y ion, its flanks and neutral losses.
+// //  PeakCombinedB2b = 8, // Represents charge 2 B ion, its flanks and neutral losses.
+// //  PeakCombinedY2b = 9, // Represents charge 2 Y ion, its flanks and neutral losses.
+// NUM_PEAK_TYPES = 10 // Total number of peak types.
 //
-// Different subclasses of TheoreticalPeakSet have varying characteristics, such
-// as the following:
-// 
-//   o  space requirements (both in memory and on disk) 
-//   o  time required to generate the peak set
-//   o  sparsity (affects space requirements and dot-product time)
-//   o  accuracy with respect to the "correct" value of the XCORR function.
-// 
-// Here are some the characteristics:
-// 
-//   TheoreticalPeakSetMakeAll           exact, slow, all peaks explicit
-//   TheoreticalPeakSetBYSparse          inexact, fast, only B and Y explicit,
-//                                       significantly larger than Diff. 
-//   TheoreticalPeakSetBYSparseOrdered   same, slightly slower to generate
-//   TheoreticalPeakSetBYAll             inexact but same as BYSparse, slow, 
-//                                       all peaks explicit
-//   TheoreticalPeakSetDiff              = MakeAll - BYAll: slow to generate,
-//                                       very sparse, completely different
-//                                       function from MakeAll. Gets stored at
-//                                       indexing and retrieved at search time.
-//   TheoreticalPeakSetSparse            For testing: very slow, explicitly
-//                                       computes both Diff and BYSparse.
+// In the current implementation (by AKF) these types for the theoretical and 
```

Comment:
Maybe list the types removed for future reference if needed.

Reply:
The list of the types are listed in the comment section: these are the PeakMain=0, LossPeak=1, etc. 
Is it ok?